iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/serviceclient/JobClientCommon.java

```diff
@@ -246,36 +289,50 @@ public void scheduleDeviceMethodSucceed() throws IOException, IotHubException, I
                 public void run()
                 {
                     String jobId = JOB_ID_NAME + UUID.randomUUID();
+                    jobIdsPending.add(jobId);
                     try
                     {
                         jobClient.scheduleDeviceMethod(
                                 jobId, queryCondition,
                                 DeviceEmulator.METHOD_LOOPBACK, RESPONSE_TIMEOUT, CONNECTION_TIMEOUT, PAYLOAD_STRING,
-                                new Date(), MAX_EXECUTION_TIME_IN_MS);
+                                new Date(), MAX_EXECUTION_TIME_IN_SECONDS);
 
                         JobResult jobResult = jobClient.getJob(jobId);
                         while(jobResult.getJobStatus() != JobStatus.completed)
                         {
                             Thread.sleep(MAXIMUM_TIME_TO_WAIT_FOR_IOTHUB);
                             jobResult = jobClient.getJob(jobId);
                         }
-                        jobResult = queryDeviceJobResult(jobId, JobType.scheduleDeviceMethod, JobStatus.completed);
-                        jobResults.put(jobId, jobResult);
+                        System.out.println("job finished with status " + jobResult.getJobStatus());
+
+                        if (jobResult.getJobStatus().equals(JobStatus.completed))
+                        {
+                            jobResult = queryDeviceJobResult(jobId, JobType.scheduleDeviceMethod, JobStatus.completed);
+                            jobResults.put(jobId, jobResult);
+                        }
+                        else
+                        {
+                            jobExceptions.put(jobId, new Exception("Scheduled job did not finish with status 'completed' but with " + jobResult.getJobStatus()));
+                        }
                     }
                     catch (IotHubException | IOException |InterruptedException e)
                     {
                         jobExceptions.put(jobId, e);
+                        System.out.println("Adding to job exceptions...");
                     }
+                    jobIdsPending.remove(jobId);
                 }
             });
         }
 
         // Cleanup
         executor.shutdown();
```

Comment:
refactor cleanup to a function for reuse

Reply:
good idea